modules/assim.batch/R/pda.define.llik.R

```diff
@@ -0,0 +1,88 @@
+##' Define PDA Likelihood Functions
+##'
+##' @title Define PDA Likelihood Functions
+##' @param all params are the identically named variables in pda.mcmc / pda.emulator
+##'
+##' @return List of likelihood functions, one for each dataset to be assimilated against.
+##'
+##' @author Ryan Kelly
+##' @export
+pda.define.llik.fn <- function(settings) {
+  # Currently just returns a single likelihood, assuming the data are flux NEE.
+  llik.fn <- list()
+  for(i in 1:length(settings$assim.batch$inputs)) {
+    # NEE + heteroskedastic Laplace likelihood
+    if(settings$assim.batch$inputs[[i]]$likelihood == "Laplace") {
+        llik.fn[[i]] <- function(model.out, obs.data, llik.par) {
+          resid <- abs(model.out - obs.data)
+          pos <- (model.out >= 0)
+          LL <- c(dexp(resid[pos], 1/(llik.par[1] + llik.par[2]*model.out[pos]), log=TRUE), 
+                  dexp(resid[!pos],1/(llik.par[1] + llik.par[3]*model.out[!pos]),log=TRUE))
+          return(list(LL=sum(LL,na.rm=TRUE), n=sum(!is.na(LL))))
+        }
+    } else {
+      # Default to Normal(0,1)
+        llik.fn[[i]] <- function(model.out, obs.data, llik.par=1) {
+          LL <- dnorm(x= obs.data, mean=model.out, sd=llik.par, log=TRUE)
```

Comment:
Yes, the rest of the workflow needs to be updated. However, when you do that you also update it so that if you use the normal likelihood you pass a standard deviation that's not 1! Indeed, that will need to be a fit parameter. That obviously doesn't need to be in the current pull request, but should be added the first time we need the Gaussian (which will be pretty soon)

Also, pda.load.data probably needs to be merged with @bcow's work on the more generic load.data